### PR TITLE
Feature Block

### PR DESCRIPTION
**Pull request by tweddielin**
This PR mainly is implementing both frontend and backend of feature distribution and linking with feature importance graph so that those two can interact with each other. By clicking the feature bar in feature importance, feature distribution will change. Pressing the toggle button will switch between training set and test set.

- Implemented FeatureBlock component as a wrapper of two sub-component, FeatureImportance and FeatureDistribution
- Made query of feature distribution for both train and test set
- Created a wrapper feature-block to include both feature importance and feature distribution
- Change metric-overtime with react-vis
- Changed the layout of model-chart
- Fixed logout button
- Changed back-to-search button css

### CODE REVIEW

File: frontend/components/featuredist-chart/component.js [hunk 1]
```diff
@@ -1,78 +1,128 @@
...
+    this.ajax_call_test()
+    this.setState({ test_button: true })
+  },
+
+  handleTrainingSet: function() {
+    this.ajax_call_train()
+    this.setState({ test_button: false })
+  },
+
+  ajax_call_test: function() {
```
Comment (thcrock) [inline frontend/components/featuredist-chart/component.js]: Is it possible to implement this with different components that share code? Most of the code in this component seems be parallel for train/test. From what I can tell, this could probably be two different components that share all code except for the URL fragment, and then the train/test button switches between the two components.

I haven't tried this myself so I don't know how easy this would be, but that's what comes to mind looking at the code

File: frontend/components/fimportance-chart/component.js [hunk 2]
```diff
@@ -33,18 +38,23 @@ export default React.createClass({
                               .sort(function(x, y) {return d3.ascending(x.x, y.x)}),
           loading: false
         })
+        self.props.handleDefaultFeatureSelected(self.state.data.slice(-1)[0].y)
```
Comment (thcrock) [inline frontend/components/fimportance-chart/component.js]: If we left off the .y here, we could use the other handler (handleValueClick or whatever) right? So we wouldn't need two separate handlers?

File: webapp/query.py [hunk 3]
```diff
@@ -211,3 +203,143 @@ def get_metrics_over_time(query_arg):
...
+    """.format(model_group_id)
+    df = pd.read_sql(
+        query,
+        con=db.engine)
+
+    feature_list = df['feature_list'].tolist()[0]
+    return feature_list
+
+
+def get_test_feature_distribution(query_arg):
```
Comment (thcrock) [inline webapp/query.py]: The code in these functions looks like it needs a little work to work for other projects. The logic seems like it's almost there, because it crawls the information_schema, so that's good. But there are two hardcoded things that should be configurable:

1. The name of the features schema (historical_features...)
2. The name of the entity_id column (officer_id) in the features tables

Could we make these configurable? Maybe require them in the database config file? They seem pretty much tied to the database information.

File: frontend/components/fimportance-chart/component.js (hunk 2 above)
Comment (tweddielin) [inline frontend/components/fimportance-chart/component.js]: yes. you're right.
